### PR TITLE
Account

### PR DESCRIPTION
**Pull request by titor-oopart**

### CODE REVIEW

File: src/main/java/com/foundation/salesforce/core/restClient/RestClientApi.java [hunk 1]
```diff
@@ -122,13 +123,24 @@ public Response patch(final String endpoint) {
      * Returns a response after requesting a post.
      *
      * @param httpMethod to do the request.
-     * @param endPoint URI.
+     * @param endPoint   URI.
```
Comment (aburgosu) [inline src/main/java/com/foundation/salesforce/core/restClient/RestClientApi.java]: delete unnecessary spaces.

File: build.gradle [hunk 2]
```diff
@@ -62,4 +63,4 @@ test {
     testLogging {
         events 'PASSED', 'STARTED', 'FAILED', 'SKIPPED'
     }
-}
+}
```
Comment (TheYisusGithub) [inline build.gradle]: Please put one salt of line.

File: src/main/java/com/foundation/salesforce/core/utils/EndPoints.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
  * @version 1.0
  */
 public final class EndPoints {
-    public static String ACCOUNT_ENDPOINT = "/services/data/v46.0/sobjects/Account";
+    public static String ACCOUNT_ENDPOINT = "/services/data/v39.0/sobjects/Account";
```
Comment (aburgosu) [inline src/main/java/com/foundation/salesforce/core/utils/EndPoints.java:22]: add final keyword to define as constants to next 4 constants

File: build.gradle (hunk 2 above)
Comment (MelissaRoman06) [inline build.gradle]: Please add the required empty line.

File: src/main/java/com/foundation/salesforce/core/restClient/RestClientApi.java (hunk 1 above)
Comment (TheYisusGithub) [inline src/main/java/com/foundation/salesforce/core/restClient/RestClientApi.java]: You have one space innecesary between to "endPoint   URI"

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+     */
+    @Given("^I fill the request with the data required$")
+    public void iFillTheRequest(Map<String, String> inputFields) {
+        accountApi.getInstance().getAccount();
+        restClientApi = RestClientApi.getInstance();
+        restClientApi.buildSpec(inputFields);
+    }
+
+    /**
+     * Creates the account
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java]: Please put point at the end line.

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+     */
+    @When("^I create an Account with the name")
+    public void iSendThePostWithTheName() {
+        response = restClientApi.post(ACCOUNT_ENDPOINT);
+        response.prettyPrint();
+        context.setResponse(response);
+    }
+
+    /**
+     * Deletes an account by id
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java]: Please put point at the end the line.

File: src/main/java/com/foundation/salesforce/core/utils/ValueAppender.java [hunk 6]
```diff
@@ -21,6 +21,7 @@
  * @author John Salazar Pinto.
  * @version 1.0
  */
+
```
Comment (MelissaRoman06) [inline src/main/java/com/foundation/salesforce/core/utils/ValueAppender.java:24]: Delete unnecessary line.

File: src/main/java/com/foundation/salesforce/entities/Account.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+package com.foundation.salesforce.entities;
+
+/**
+ * Account class.
+ *
+ * @author John Salazar Pinto
+ * @version 1.0
+ */
+public class Account {
+
```
Comment (MelissaRoman06) [inline src/main/java/com/foundation/salesforce/entities/Account.java]: Delete unnecessary line.

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+import static com.foundation.salesforce.core.utils.EndPoints.ACCOUNT_ENDPOINT;
+
+/**
+ * AccountStep class with the steps for the account feature
+ *
+ * @author John Salazar Pinto
+ * @version 1.0
+ */
+public class AccountStep {
+
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java]: Delete unnecessary line.

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+     * Deletes an account by id
+     */
+    @When("^I delete an account that previusly was created$")
+    public void iFillTheDeleteRequest() {
+        restClientApi = RestClientApi.getInstance();
+        response = restClientApi.delete(ACCOUNT_ENDPOINT + "/" + context.getAccount().getId());
+        response.prettyPrint();
+    }
+
+    @Then("I delete the account that previously was created is {int}")
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java]: Please add documentation.

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+    public void iDeleteTheAccountThatPreviouslyWasCreatedIs(int statusCode) {
+        json = response.then().statusCode(statusCode);
+        Assert.assertEquals(response.getStatusCode(), statusCode);
+    }
+
+    @Then("the status code is a number {int}")
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java:110]: Please add documentation.

File: src/main/java/com/foundation/salesforce/core/api/AccountApi.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    /**
+     * Variable for the response.
+     */
+    private Response response;
+
+    /**
+     * Constructor of AccountAPI.
+     */
+    protected AccountApi() {
+            restClient = RestClientApi.getInstance();
```
Comment (MelissaRoman06) [inline src/main/java/com/foundation/salesforce/core/api/AccountApi.java]: Please correct indentation.

File: src/main/java/com/foundation/salesforce/core/restClient/RestClientApi.java (hunk 1 above)
Comment (MelissaRoman06) [inline src/main/java/com/foundation/salesforce/core/restClient/RestClientApi.java]: Is that blank space necessary?

File: src/main/java/com/foundation/salesforce/core/restClient/RestClientApi.java [hunk 12]
```diff
@@ -122,13 +123,24 @@ public Response patch(final String endpoint) {
...
      * @return a RestAssured Response structure.
      */
     public Response apiResponse(String httpMethod, final String endPoint) {
         return given().spec(request).when().request(httpMethod, endPoint);
     }
 
+    /**
+     * Returns a response after requesting a post.
+     * @param endpoint URI.
```
Comment (MelissaRoman06) [inline src/main/java/com/foundation/salesforce/core/restClient/RestClientApi.java:136]: Please add an empty line according to our code conventions.

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+import org.json.JSONObject;
+import io.restassured.response.Response;
+import org.testng.Assert;
+
+import java.util.Map;
+
+import static com.foundation.salesforce.core.utils.EndPoints.ACCOUNT_ENDPOINT;
+
+/**
+ * AccountStep class with the steps for the account feature
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java]: Please put a point at the end the this line.

File: src/main/java/com/foundation/salesforce/core/api/AccountApi.java [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+ * with Jala Foundation.
+ */
+
+package com.foundation.salesforce.core.api;
+
+import com.foundation.salesforce.core.restClient.Authentication;
+import com.foundation.salesforce.core.restClient.RestClientApi;
+import io.restassured.response.Response;
+
+import static com.foundation.salesforce.core.utils.EndPoints.ACCOUNT_ENDPOINT;
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/core/api/AccountApi.java:17]: Please, place this line in the same group, i.e. after import com.foudnation.salesforce.core.restClient. ... group

File: src/test/java/com/foundation/salesforce/hooks/AccountHooks.java [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+package com.foundation.salesforce.hooks;
+
+import com.foundation.salesforce.core.restClient.Authentication;
+import com.foundation.salesforce.core.restClient.RestClientApi;
+import com.foundation.salesforce.core.utils.EndPoints;
+import com.foundation.salesforce.core.utils.ValueAppender;
+import com.foundation.salesforce.entities.Context;
+import cucumber.api.java.After;
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/hooks/AccountHooks.java:21]: Place a separator blank line.

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+ */
+
+package com.foundation.salesforce.steps;
+
+import com.foundation.salesforce.core.api.AccountApi;
+import com.foundation.salesforce.core.restClient.RestClientApi;
+import com.foundation.salesforce.core.utils.EndPoints;
+import com.foundation.salesforce.core.utils.ResponseValidation;
+import com.foundation.salesforce.entities.Context;
+import cucumber.api.java.en.And;
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java:21]: Please, place a separator blank line.

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import io.restassured.response.ValidatableResponse;
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java:25]: Please, place a separator blank line.

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+import io.restassured.specification.RequestSpecification;
+import org.json.JSONObject;
+import io.restassured.response.Response;
+import org.testng.Assert;
+
+import java.util.Map;
+
+import static com.foundation.salesforce.core.utils.EndPoints.ACCOUNT_ENDPOINT;
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java]: Please, move this line to the corresponding group.

**Review by alszla — APPROVED** [review]

File: src/test/java/com/foundation/salesforce/steps/AccountStep.java [hunk 19]
```diff
@@ -105,12 +102,10 @@ public void iFillTheDeleteRequest() {
...
 
-    @Then("I delete the account that previously was created is {int}")
-    public void iDeleteTheAccountThatPreviouslyWasCreatedIs(int statusCode) {
-        json = response.then().statusCode(statusCode);
-        Assert.assertEquals(response.getStatusCode(), statusCode);
-    }
-
+    /**
+     * Verifies response's status code.
+     * @param statusCode Expected status for delete account
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/AccountStep.java]: Please add an empty line above.

**Review by MelissaRoman06 — APPROVED** [review]

**Pull request merged**